lib/addressable/uri.rb

```diff
@@ -74,6 +74,9 @@ module CharacterClasses
       "ldap" => 389,
       "prospero" => 1525
     }
+    ##
+    #  @@avoid_host_omitted is to avoid host validate when is set to be omitted
+    @@avoid_host_omitted = true
```

Comment:
Why add this?

Reply:
Because, when the method "omit" is called, the host is nullified, and then call the method  "validate" the error `InvalidURIError, "Hostname not supplied:`. So I don't link this approach to use @@ in this case, but I don't see any solutions to solve this problem without a huge refactory. Do you have any idea?